cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js

```diff
@@ -73,93 +73,98 @@ const visitPage = (service, pageType) => {
+        .forEach(pageType => {
+          const { variant } = config[service];
+
+          describe(`Canonical Cookie Banner Test for ${service} ${pageType}`, () => {
+            it('should have a privacy & cookie banner, which disappears once "accepted" ', () => {
+              cy.clearCookies();
+              visitPage(service, pageType);
+
+              getPrivacyBanner(service, variant).should('be.visible');
+              getCookieBanner(service, variant).should('not.be.visible');
+
+              assertCookieValues({
+                ckns_privacy: 'july2019',
+                ckns_policy: '000',
+              });
+
+              getPrivacyBannerAccept(service, variant).click();
+
+              getCookieBanner(service, variant).should('be.visible');
+              getPrivacyBanner(service, variant).should('not.be.visible');
+
+              getCookieBannerAccept(service, variant).click();
+
+              assertCookieValues({
+                ckns_explicit: '1',
+                ckns_privacy: 'july2019',
+                ckns_policy: '111',
+              });
+
+              getCookieBanner(service, variant).should('not.be.visible');
+              getPrivacyBanner(service, variant).should('not.be.visible');
+
+              ensureCookieExpiryDates();
             });
 
-            getPrivacyBannerAccept(service, variant).click();
+            it('should have a privacy banner that disappears once accepted but a cookie banner that is rejected', () => {
+              cy.clearCookies();
+              visitPage(service, pageType);
 
-            getCookieBanner(service, variant).should('be.visible');
-            getPrivacyBanner(service, variant).should('not.be.visible');
+              getPrivacyBanner(service, variant).should('be.visible');
+              getCookieBanner(service, variant).should('not.be.visible');
 
-            getCookieBannerAccept(service, variant).click();
+              assertCookieValues({
+                ckns_privacy: 'july2019',
+                ckns_policy: '000',
+              });
 
-            assertCookieValues({
-              ckns_explicit: '1',
-              ckns_privacy: 'july2019',
-              ckns_policy: '111',
-            });
-
-            getCookieBanner(service, variant).should('not.be.visible');
-            getPrivacyBanner(service, variant).should('not.be.visible');
+              getPrivacyBannerAccept(service, variant).click();
+              getCookieBannerReject(service, variant).click();
 
-            ensureCookieExpiryDates();
-          });
+              visitPage(service, pageType);
 
-          it('should have a privacy banner that disappears once accepted but a cookie banner that is rejected', () => {
-            cy.clearCookies();
-            visitPage(service, pageType);
+              assertCookieValues({
+                ckns_explicit: '1',
+                ckns_privacy: 'july2019',
+                ckns_policy: '000',
```

Comment:
What do these assertions mean?

Reply:
it uses this method: https://github.com/bbc/simorgh/blob/71b50d1c65d33bc4baebbb8e7066c2e3d26067ca/cypress/integration/specialFeatures/cookieBanner/testsForCanonicalOnly.js#L64-L68

it checks the cookie (ckns_explicit, ckns_privacy, ckns_policy) has the corresponding value assigned to it.

in this case `ckns_explicit ` has been accepted so the value is `1` and cookie banner was rejected so the value is `000`.

Reply:
(These tests already existed, the only change I made is here: https://github.com/bbc/simorgh/pull/4391/files#diff-b516ab6e536a3fd977362f8719c08543R76-R79)

github made it looked weird 🤕